### PR TITLE
Entites are now component based!

### PR DESCRIPTION
**Pull request by Repertoi-e**
Have fun reviewing this :) lol

### CODE REVIEW

File: src/states/StatePlaying.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace State
 		m_level.addEntity(&m_player);
 		m_level.player = &m_player;
 
-		m_level.addEntity(new Framework::Zombie());
```
Comment (Hopson97) [inline src/states/StatePlaying.cpp]: Please do not use raw pointers :) (for ownership)

File: src/entity/Entity.cpp [hunk 2]
```diff
@@ -1,56 +1,26 @@
...
-	{
-		m_health -= dmg.amount;
-	}
-
-	int32 Entity::getHealth()
+	Entity::Entity(sf::Vector2f& position, sf::Sprite& sprite)
+	: m_ID(Random::uint64InRange(0, 18446744073709551614))
 	{
-		return m_health;
+		addComponent(new PositionComponent(position));
```
Comment (Hopson97) [inline src/entity/Entity.cpp]: Please do not use raw pointers :) (for ownership)

File: src/entity/Entity.cpp [hunk 3]
```diff
@@ -1,56 +1,26 @@
...
+		addComponent(new SpriteComponent(sprite));
```
Comment (Hopson97) [inline src/entity/Entity.cpp]: Please do not use raw pointers :) (for ownership)

**Comment by Repertoi-e** [comment]
I'll do it tomorrow too tired now.

**Pull request merged**